src/tribler/core/conftest.py

```diff
@@ -64,6 +65,13 @@ def pytest_runtest_protocol(item: Function, log=True, nextitem=None):
         yield
 
 
+@pytest.fixture(autouse=True)
+def ensure_gc():
+    """ Ensure that the garbage collector runs after each test.
+    This is critical for test stability as we use Libtorrent and need to ensure all its destructors are called. """
+    gc.collect()
```

Comment:
Currently, the fixture triggers the garbage collection at the beginning of the (next) test. For that reason, the errors triggered during the garbage collection phase will take place not in the erroneous test but in the randomly scheduled next test. Usually, these errors are silently suppressed, as any exception in `__del__` methods is silently suppressed, but they still can somehow affect the test, I think.

By adding the `yield` we can move the garbage collection phase to the end of the current test, to not affect the next test:

```suggestion
    yield
    gc.collect()
```

Reply:
Applied.